### ISSUE TITLE
DNSMASQ_LISTENING envvar not working in 2022.08 and above

### ISSUE THREAD

**Issue by darkrain42** [pi-hole/docker-pi-hole#1188]
This is a: Bug.

## Details

In 2022.08 and 2022.08.2, the `DNSMASQ_LISTENING` environment variable is broken / doesn't work.  Pi-hole starts up with the DNS interface settings set to "Allow only local requests".

This appears (to me) to have been broken by #1141 (namely 611df9b3f08fa5c421ab6eca980994dd8ba9a44a).  (At least, that is the last commit where I can see a reference to the environment variable.)

Works as expected in 2022.07:
```shell
paul@vaarsuvius ~ % docker run --rm -it -d -e DNSMASQ_LISTENING=all --name pihole pihole/pihole:2022.07
becace59b7562cf4b6ce3cd30237151d5b72d3fc98f03622c5cd24c88a110782
paul@vaarsuvius ~ % docker exec -i pihole grep except /etc/dnsmasq.d/01-pihole.conf
except-interface=nonexisting
paul@vaarsuvius ~ % docker kill pihole
pihole
```

Broken in 2022.08 and 2022.08.2:
```shell
paul@vaarsuvius ~ % docker run --rm -it -d -e DNSMASQ_LISTENING=all --name pihole pihole/pihole:2022.08
26aad1012f9e8a80d30f62b67bf03b66878623d39ff1502bcb232fc13165e1ec
paul@vaarsuvius ~ % sleep 30; docker exec -i pihole grep except /etc/dnsmasq.d/01-pihole.conf
paul@vaarsuvius ~ % docker kill pihole
pihole
paul@vaarsuvius ~ % docker run --rm -it -d -e DNSMASQ_LISTENING=all --name pihole pihole/pihole:2022.08.2
f8e21b73edfd1b94d7e3695ae4da39ab383cc1085dafabb99dd39064d46f9f04
paul@vaarsuvius ~ % sleep 10; docker exec -i pihole grep except /etc/dnsmasq.d/01-pihole.conf
paul@vaarsuvius ~ %
```

## Related Issues
- [X] I have searched this repository/Pi-hole forums for existing issues and pull requests that look similar 

## How to reproduce the issue 

1. Environment data
  * Operating System: Ubuntu 22.04.1
  * Hardware: PC
  * Kernel Architecture: amd64
  * Docker Install Info and version: docker 20.10.12-0ubuntu4
    - Software source: Ubuntu-provided package
  * Hardware architecture: also amd64 :)

2. docker run --rm -it -d -e DNSMASQ_INTERFACES=all pihole/pihole    # (and then examine /etc/dnsmasq.d/01-pihole.conf in the container)

## These common fixes didn't work for my issue
- [X] I have tried removing/destroying my container, and re-creating a new container
- [X] I have tried fresh volume data by backing up and moving/removing the old volume data
- [x] I have tried running the stock `docker run` example(s) in the readme (removing any customizations I added)
- [X] I have tried a newer or older version of Docker Pi-hole (depending what version the issue started in for me)
- [X] I have tried running without my volume data mounts to eliminate volumes as the cause

**Comment by rdwebdesign** [comment]
What happens if you try to set via Web Interface?
![image](https://user-images.githubusercontent.com/1385443/187055742-27ff0508-0f06-4b40-a44a-539b46b6c54d.png)

**Comment by darkrain42** [comment]
Setting it in the GUI works, yes.

```
paul@vaarsuvius ~ % docker run --rm -it -d -e DNSMASQ_LISTENING=all -e WEBPASSWORD=password --name pihole pihole/pihole:2022.08.2
6134e9a0108003b4d8568ed6e9fef2cbb88cb3683b9f408b9bb9b2d1931650b3
paul@vaarsuvius ~ % docker exec -i pihole grep except /etc/dnsmasq.d/01-pihole.conf  # before setting in GUI
paul@vaarsuvius ~ % docker exec -i pihole grep except /etc/dnsmasq.d/01-pihole.conf  # after setting in GUI 
except-interface=nonexisting
paul@vaarsuvius ~ % 
```

**Comment by PromoFaux** [comment]
Apologies - not sure how I managed to remove this completely.. will look at adding it back!

**Comment by PromoFaux** [comment]
Fixed in `2022.08.3` which will have finished building in about 10 mins
